xmipp

```diff
@@ -677,54 +738,59 @@ int main(){}
+               configDict["MPI_CXXFLAGS"])):
+        print(red("Check the INCDIRFLAGS, MPI_CXX and CXXFLAGS in 'xmipp.conf'"))
+        print(red("hint: MPI_CXXFLAGS is normally not needed, but if it is might help 'mpicc --showme:compile'"))
         return False
     libhdf5=getHdf5Name(configDict["LIBDIRFLAGS"])
-    if not runJob("%s %s %s xmipp_mpi_test_main.o -o xmipp_mpi_test_main -lfftw3 -lfftw3_threads -l%s  -lhdf5_cpp -ltiff -ljpeg -lsqlite3 -lpthread" % \
-              (configDict["MPI_LINKERFORPROGRAMS"], configDict["LINKFLAGS"], configDict["LIBDIRFLAGS"], libhdf5)):
+    if not runJob("%s %s %s %s xmipp_mpi_test_main.o -o xmipp_mpi_test_main "
+                  "-lfftw3 -lfftw3_threads -l%s  -lhdf5_cpp -ltiff -ljpeg -lsqlite3 -lpthread" % \
+              (configDict["MPI_LINKERFORPROGRAMS"], configDict["LINKFLAGS"],
+               configDict["MPI_LINKFLAGS"], configDict["LIBDIRFLAGS"], libhdf5)):
         print(red("Check the LINKERFORPROGRAMS, LINKFLAGS and LIBDIRFLAGS"))
+        print(red("hint: MPI_LINKFLAGS is normally not needed, but if it is might help 'mpicc --showme:link'"))
         return False
     runJob("rm xmipp_mpi_test_main*")
 
-    ok = False
     echoString = blue("   > This sentence should be printed 4 times if mpi runs fine")
-    if checkProgram("mpirun",False):
-        ok=(runJob("mpirun -np 4 echo '%s (by mpirun).'" % echoString) or
-            runJob("mpirun -np 4 --allow-run-as-root echo '%s (by mpirun).'" % echoString))
-    elif checkProgram("mpiexec",False):
-        ok=(runJob("mpiexec -np 4 echo '%s (by mpiexec).'" % echoString) or
-            runJob("mpiexec -np 4 --allow-run-as-root echo '%s (by mpiexec).'" % echoString))
-    else:
+    if not (runJob("%s -np 4 echo '%s.'" % (configDict['MPI_RUN'], echoString)) or
+            runJob("%s -np 4 --allow-run-as-root echo '%s.'" % (configDict['MPI_RUN'], echoString))):
         print(red("mpirun or mpiexec have failed."))
-
-    return ok
+        return False
+    return True
 
 
 def configJava(configDict):
     if configDict["JAVA_HOME"]=="":
-        javaProgramPath = distutils.spawn.find_executable("java")
-        javaHomeDir = None
+        javaProgramPath = whereis('java', findReal=True)
+        if not javaProgramPath:
+            javaProgramPath = findFileInDirList('java', ['/usr/lib/jvm/java-*/bin'])  # put candidates here
         if javaProgramPath:
-            javaProgramPath=os.path.dirname(os.path.realpath(javaProgramPath))
-            javaHomeDir = javaProgramPath.replace("/jre/bin","")
-            javaHomeDir = javaHomeDir.replace("/bin","")
-        if javaHomeDir:
-            configDict["JAVA_HOME"]=javaHomeDir
-            print(green("%s detected." % os.path.basename(javaHomeDir)))
-
-    if configDict["JAVA_BINDIR"]=="" and javaHomeDir:
+            updateXmippEnv(PATH=javaProgramPath)
+            javaHomeDir = javaProgramPath.replace("/jre/bin", "")
+            javaHomeDir = javaHomeDir.replace("/bin", "")
```

Comment:
Not really. `javaHomeDir` is often  `.........java-8/bin`, but some times it is  `....../java-8/jre/bin`. If it's the first, L711 does nothing; if it's the second, L712 does nothing. They are alternative, not complementary. 

I thing it can be better, but I've rearranged this couple of lines, but they was before. I can improve it, I'll try.